src/main/java/com/trolley/trolley/InvoiceLineGateway.java

```diff
@@ -0,0 +1,184 @@
+package com.trolley.trolley;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trolley.Exceptions.InvalidFieldException;
+
+public class InvoiceLineGateway
+{
+    Client client;
+    
+    public InvoiceLineGateway(final Configuration config) {
+        this.client = new Client(config);
+    }
+    
+    /**
+     * Create a single Invoice Line and add it to the Invoice whose ID is supplied. <br/>
+     * 
+     * @param invoiceId Id of the invoice where this line will be added
+     * @param invoiceLine InvoiceLine object which will be created
+     * @return Invoice object of the updated invoice containing the newly created line
+     * @throws Exception
+     */
+    public Invoice create(final String invoiceId, final InvoiceLine invoiceLine) throws Exception {
+        final String endPoint = "/v1/invoices/create-lines/";
+        ArrayList<InvoiceLine> invoiceLines = new ArrayList<InvoiceLine>() {
+            {
+                add(invoiceLine);
+            }
+        };
+
+        final String body = "{"
```

Comment:
In general - we should never serialize/deserialize objects like this. We should use either built-in or industry standard ways to achieve this. I believe in this case `Jackson mapper` should help.